client/coral-framework/services/plugins.js

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import uniq from 'lodash/uniq';
+import pick from 'lodash/pick';
+import merge from 'lodash/merge';
+import flattenDeep from 'lodash/flattenDeep';
+import isEmpty from 'lodash/isEmpty';
+import flatten from 'lodash/flatten';
+import mapValues from 'lodash/mapValues';
+import {getDisplayName} from 'coral-framework/helpers/hoc';
+import camelize from '../helpers/camelize';
+import uuid from 'uuid/v4';
+
+// This is returned for pluginConfig when it is empty.
+const emptyConfig = {};
+
+// Memoize the warnings so we only show them once.
+const memoizedWarnings = [];
+
+// withWarnings decorates the props of queryData with a proxy that
+// prints a warning when accessing deeper props.
+function withWarnings(component, queryData) {
+  if (process.env.NODE_ENV !== 'production' && window.Proxy) {
+
+    // Show warnings when accessing queryData only when not in production.
+    return mapValues(queryData, (value, key) => {
+
+      // Keep null values..
+      if (!queryData[key]) {
+        return queryData[key];
+      }
+      return new Proxy(queryData[key], {
+        get(target, name) {
+
+          // Detect access from React DevTools and ignore those.
+          const error = new Error();
+          const accessFromDevTools = ['backend.js', 'dehydrate']
+            .every((keyword) => error.stack && error.stack.includes(keyword));
+
+          // Only care about the components defined in the plugins.
+          if (component.talkPluginName && !accessFromDevTools) {
+            const warning = `'${getDisplayName(component)}' of '${component.talkPluginName}' accessed '${key}.${name}' but did not define fragments using the withFragment HOC`;
+            if (memoizedWarnings.indexOf(warning) === -1) {
+              console.warn(warning);
+              memoizedWarnings.push(warning);
+            }
+          }
+          return queryData[key][name];
+        }
+      });
+    });
+  }
+
+  return queryData;
+}
+
+function addMetaDataToSlotComponents(plugins) {
+
+  // Add talkPluginName to Slot Components.
+  plugins.forEach((plugin) => {
+    const slots = plugin.module.slots;
+    slots && Object.keys(slots).forEach((slot) => {
+      slots[slot].forEach((component) => {
+
+        // Attach plugin name to the component
+        component.talkPluginName = plugin.name;
+
+        // Attach uuid to the component
+        component.talkUuid = uuid();
```

Comment:
@wyattjoh I'd love to do that, because this is my own personal preference too, but at least on the frontend side we have not been following this convention. Like `URL` in `baseUrl` and `buildUrl` or `URI` in `redirectUri`. That made me think that our convention was different.

If this is our convention we should make another PR and go through our code and make sure we follow it.